tool/templates/cpp/method_decl.h.jinja

```diff
@@ -8,4 +8,12 @@ inline {##}
 		{{ param.type_name }} {{ param.var_name }}
 	{%- endfor -%}
 )
-{%- for qualifier in m.post_qualifiers %} {{qualifier}}{% endfor %};
\ No newline at end of file
+{%- for qualifier in m.post_qualifiers %} {{qualifier}}{% endfor %};
+{#- Extra method definitions for special types -#}
+{%- match m.method.attrs.special_method -%}
+{%- when Some(hir::SpecialMethod::Iterable) -%}
+{%- let helper_type = m.return_ty.replace("std::unique_ptr", "diplomat::next_to_iter_helper") %}
```

Comment:
thought: ideally we're not string replacing on types, we could potentially instead return `make_helper<T>` where `make_helper` does the appropriate transform on types that are unique_ptrs via specialization